backend/src/main/java/ai/verta/modeldb/App.java

```diff
@@ -341,4 +325,42 @@ public static void initializeTelemetryBasedOnConfig(MDBConfig mdbConfig)
       LOGGER.info("Telemetry opt out by user");
     }
   }
+
+  @PreDestroy
+  public void onShutDown() {
+    if (App.getInstance().server.isPresent()) {
+      var server = App.getInstance().server.get();
+      try {
+        int activeRequestCount = MonitoringInterceptor.ACTIVE_REQUEST_COUNT.get();
+        while (activeRequestCount > 0) {
+          activeRequestCount = MonitoringInterceptor.ACTIVE_REQUEST_COUNT.get();
+          LOGGER.info("Active Request Count in while:{} ", activeRequestCount);
+          Thread.sleep(1000); // wait for 1s
+        }
```

Comment:
This while loop has no timeout, remove it. 
```suggestion
```

Reply:
We can not remove it. it is check active request for GRPC services. I will improve code with some interval logic which break while loop like you have did in GracefulShutdown.java